src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+function createMessage(text, isError = false) {
+  const messageDiv = document.createElement('div');
+
+  messageDiv.dataset.qa = 'notification';
+
+  if (!isError) {
+    messageDiv.classList.add('success');
+  } else {
+    messageDiv.classList.add('warning');
+  }
+
+  messageDiv.textContent = text;
+  document.body.appendChild(messageDiv);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', function() {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', function() {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let rightClick = false;
+  let leftClick = true;
+
+  document.addEventListener('click', function() {
```

Comment:
This could be shortened, instead of registering two events, one on click and the other on the contextmenu you could register only one `mousedown` and based on the [button](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button) property on the event change leftClick and rightClick:
```js
body.addEventListener('mousedown', ({ button }) => {
  if (button === 0) {
    left = true;
  }

  if (button === 2) {
    right = true;
  }

  if (left && right) {
    resolve('Third promise was resolved');
  }
});
```

Reply:
Oh, that's a nice solution. Thank you! :)